orbit/pkg/useraction/mdm_migration_darwin.go

```diff
@@ -338,21 +349,22 @@ func (m *swiftDialogMDMMigrator) waitForUnenrollment() error {
 }
 
 func (m *swiftDialogMDMMigrator) renderMigration() error {
-	log.Debug().Msg("checking manual enrollment status")
-	manualProfileCheck, err := profiles.IsManuallyEnrolledInMDM()
+	log.Debug().Msg("checking current enrollment status")
+	isCurrentlyManuallyEnrolled, err := profiles.IsManuallyEnrolledInMDM()
 	if err != nil {
 		return err
 	}
 
-	// Check if we're in a manual migration.
-	migrationType, err := m.mrw.GetMigrationType()
+	// Check what kind of migration was in progress, if any.
+	previousMigrationType, err := m.mrw.GetMigrationType()
 	if err != nil {
 		log.Error().Err(err).Msg("getting migration type")
 	}
 
-	isManual := manualProfileCheck || migrationType == constant.MDMMigrationTypeManual
+	m.props.IsManualMigration = isCurrentlyManuallyEnrolled || previousMigrationType == constant.MDMMigrationTypeManual
+	m.props.IsADEMigration = !isCurrentlyManuallyEnrolled || previousMigrationType == constant.MDMMigrationTypeADE
```

Comment:
not a big deal: in the "react paradigm" anything that's props shouldn't be modified internally by the "component", it was a bit confusing to me at first because I couldn't see where the caller was passing these.

if we need to keep internal state, why not use an internal property?

Reply:
yeah, I realized that this morning. I could pull this out of the function, but that'd be more work to test. I'm just going to use some local variables and pass those to the `getMessageAndFlags` method.